package.json

```diff
@@ -137,6 +137,7 @@
 		"gulp-sass": "4.1.0",
 		"gulp-sourcemaps": "2.6.5",
 		"gulp-tap": "2.0.0",
+		"htm": "3.0.4",
```

Comment:
I just noticed that we're already using preact on jetpack, I had no idea!!

Maybe switching to JSX for the user facing part of the block makes more sense now... `htm` has one advantage though, I can be exposed to users so they can customize the views on their end. For instance [here](https://github.com/Automattic/jetpack/pull/16302/files#diff-a1ecaecd7e92d91bdb9f51c729281dc7) I defined the default views that can be customized. Sounds like a YAGNI though now that we have preact with JSX built in 🤔

Reply:
We don't use Preact in the extensions, though; it's limited to a small part of the codebase (the Instant Search interface), so you wouldn't be able to rely on Preact for your block I'm afraid.

Reply:
Oh I see! The part that uses preact here isn't really part of the block actually and could be moved out of the extension if we want to. It's just the visitor-facing UI that renders the story on the site. Preact I'm pretty sure of as it fits our requirements for a lightweight UI framework.

Reply:
Thanks for the extra details, that helps!

It does sound like Preact could indeed do the job. However, I am not sure it would be the best choice on the long term here. We've tried to stay as close as possible to Core with blocks so far:
- it allows us to benefit from the work the community does every time a component is created and updated.
- it integrates well with existing components and processes such as translations.
- it helps us making sure that whatever component updates will happen in the future can be adapted to Jetpack components.
- it makes it easier to onboard folks creating new blocks or picking up maintenance of a block.
- it allows us to benefit from existing building tools as well.

Moving away from those things isn't something I would personally recommend, as it can impact how the block may be maintained and involve in the future.

To add a bit of background, we picked Preact for the Instant Search interface because that's something that needs to be loaded on each and every page of a site. In this case, the size of the bundle we serve is extremely important, and after some tests we concluded that using Preact over WordPress components was worth it, even if we had to go through some loops to implement things such as i18n.

Reply:
Including Preact and related libraries in bundle here _and_ in Search doubles up KBs it uses, too. ;-)